### ISSUE TITLE
aligning items to the right of the container

### ISSUE THREAD

**Issue by BurningTreeC** [haltu/muuri#255]
Hi there, I'm using your library for a [tiddlywiki](https://tiddlywiki.com) layout which can be installed as a plugin - it soon will have a repository here on github

There's one minor issue I have when using the built-in horizontal layout in combination with `alignRight=true`. The container div has `min-width: 100%;` and I'd like to align the items to the very right of that container. Currently they stay aligned to the left of the container, only the sorting is done right-to-left.

Is there an easy way to change that behavior?

I understand I could modify the code that sets the `translateX()` value and add the difference to the right of the container (if there's no overflow) but I'd prefer a way without touching your library

all the best,
Simon

**Comment by niklasramo** [comment]
Can you create a reduced test case for that scenario in e.g. codepen? Would help me understand the problem better.

**Comment by BurningTreeC** [comment]
Hi @niklasramo , thanks for answering!

See [this pen](https://codepen.io/burningtreec/pen/vbrRPz)

... it's horizontal and right-aligned. as long as there's no overflow to the right, items seem to be aligned to the left (even if the sorting is right-to-left). I'd like to have those items at the right, so that they "dock" to the right edge of the page, exactly like the picture if you remove the line `horizontal: true`

**Comment by niklasramo** [comment]
Thanks for the live demo! So the way to solve the problem is aligning the container (grid) to to the right side.

A crude example with using 'float: right': https://codepen.io/anon/pen/xMJLEp

**Comment by BurningTreeC** [comment]
@niklasramo , big thanks, that works! love it when solutions are that simple, I didn't even think about doing this :+1: 

I'll leave a link to the repo here if you're interested, there will be a page with the demo-layout

**Comment by BurningTreeC** [comment]
I changed this solution from using `float` to `margin-left: auto; margin-right: smallvalue;` which is working better in my case.

still, I figured muuri isn't made for this ... it doesn't work correctly if the container resizes while dragging, it results in "jumping" of the items so that they are no-more positioned below the cursor.
this all is just about the edge-case where the container width is lower than the viewport width

I tried some workarounds using a MutationObserver I already have listening on the container for other reasons, which allows me to update the positions of the items relative to the changed width of the container and update the items with new positions on dragmove, too, but it's too cumbersome to do it that way

I figured that the best would be a built-in switch that makes muuri add the container's left margin to the translateX values (for horizontal + alignRight only)

As this is an edge-case and I can live without this edge-case behaving like I'd like it to do, I can close this issue, but thanks @niklasramo for your help!
